crates/ruma-client-api/src/r0/session/fallback.rs

```diff
@@ -0,0 +1,52 @@
+//! [GET /_matrix/static/client/login/](https://spec.matrix.org/unstable/client-server-api/#login-fallback)
+
+use ruma_api::ruma_api;
+use ruma_identifiers::DeviceId;
+
+ruma_api! {
+    metadata: {
+        description: "Get login fallback web page.",
+        method: GET,
+        name: "login_fallback",
+        path: "/_matrix/static/client/login/",
+        rate_limited: false,
+        authentication: None,
+    }
+
+    request: {
```

Comment:
Since all fields are optional, it would be good to have a `Default` implementation.

```suggestion
    #[derive(Default)]
    request: {
```